Gruntfile.js

```diff
@@ -1,63 +1,187 @@
+      options: {
+        files: ['package.json', 'bower.json'],
+        updateConfigs: ['pkg'],
+        pushTo: "<%= grunt.config.get('gh-release.remote') %>",
+        commitFiles: ['package.json', grunt.config.get('changelog'), 'bower.json', 'dist']
+      }
+    },
+
+    prompt: {
+      'gh-release': {
+        options: {
+          questions: [
+            {
+              config: 'gh-release.remote',
+              type: 'input',
+              message: 'Git Remote (usually GoogleChrome or origin)',
+              default: 'GoogleChrome',
+              validate: function(val) {
+                return (grunt.util._.size(val) > 0);
+              }
+            },
+            {
+              config: 'gh-release.repo',
+              type: 'input',
+              message: 'Github Repository',
+              default: 'GoogleChrome/accessibility-developer-tools',
+              validate: function(val) {
+                return (grunt.util._.size(val) > 0);
+              }
+            },
+            {
+              config: 'gh-release.username',
+              type: 'input',
+              message: 'Github Username',
+              validate: function(val) {
+                return (grunt.util._.size(val) > 0);
+              }
+            },
+            {
+              config: 'gh-release.password',
+              type: 'password',
+              message: 'Github Password or Token',
+              validate: function(val) {
+                return (grunt.util._.size(val) > 0);
+              }
+            }
+          ]
+        }
+      }
     }
   });
 
-  grunt.loadNpmTasks('grunt-closurecompiler');
-  grunt.loadNpmTasks('grunt-contrib-qunit');
+  grunt.registerTask('changelog', function(type) {
+    grunt.task.requires('bump-only:' + type);
 
-  grunt.registerTask('git-describe', function() {
-    var _spawn = require("grunt-util-spawn")(grunt);
+    var config = {
+      data: {
+        version: grunt.config.get('pkg.version'),
+        releaseDate: grunt.template.today("yyyy-mm-dd")
+      }
+    };
 
+    var stopRegex = /^\#\#\ [0-9]+.*$/m;
+    var stopIndex = 0;
+    var releaseNotes = '';
+    var dest = grunt.config.get('changelog');
+    var contents = grunt.file.read(dest);
+    var headerTpl = "## <%= version %> - <%= releaseDate %>\n\n";
+    var header = grunt.template.process(headerTpl, config);
+
+    if (contents.length > 0) {
+      if ((stopIndex = contents.search(stopRegex)) !== -1) {
+        releaseNotes = contents.slice(0, stopIndex);
+      }
+    }
+
+    grunt.config.set("gh-release.release-notes", releaseNotes);
+
+    grunt.file.write(dest, "" + header + contents);
+    grunt.log.ok("Changelog updated, and release notes extracted.");
+  });
+
+  grunt.registerTask('gh-release', function() {
+    var config = grunt.config.get('gh-release');
+    var pkg = grunt.config.get('pkg');
+    var done = this.async();
+
+    request
+      .post('https://api.github.com/repos/' + config.repo + '/releases')
```

Comment:
The only scenario I can think of where you'd query the github API with a different base URL would be if you deployed an internal github enterprise system. If that's in the cards, we can certainly set this up as a config option.

That specific grunt task is being refactored in my experimental branch though, after discussions with @alice regarding how she'd like github releases to be managed.